domain/include/cstone/sfc/box.hpp

```diff
@@ -136,41 +149,37 @@ class Box
     HOST_DEVICE_FUN constexpr T ily() const { return inverseLengths_[1]; }
     HOST_DEVICE_FUN constexpr T ilz() const { return inverseLengths_[2]; }
 
-    HOST_DEVICE_FUN constexpr bool pbcX() const { return pbc[0]; } // NOLINT
-    HOST_DEVICE_FUN constexpr bool pbcY() const { return pbc[1]; } // NOLINT
-    HOST_DEVICE_FUN constexpr bool pbcZ() const { return pbc[2]; } // NOLINT
+    HOST_DEVICE_FUN constexpr int boundaryX() const { return boundaries[0]; } // NOLINT
+    HOST_DEVICE_FUN constexpr int boundaryY() const { return boundaries[1]; } // NOLINT
+    HOST_DEVICE_FUN constexpr int boundaryZ() const { return boundaries[2]; } // NOLINT
+
+    HOST_DEVICE_FUN constexpr bool pbcX() const { return boundaries[0] == 1; } // NOLINT
+    HOST_DEVICE_FUN constexpr bool pbcY() const { return boundaries[1] == 1; } // NOLINT
+    HOST_DEVICE_FUN constexpr bool pbcZ() const { return boundaries[2] == 1; } // NOLINT
+
+    HOST_DEVICE_FUN constexpr bool fbcX() const { return boundaries[0] == 2; } // NOLINT
+    HOST_DEVICE_FUN constexpr bool fbcY() const { return boundaries[1] == 2; } // NOLINT
+    HOST_DEVICE_FUN constexpr bool fbcZ() const { return boundaries[2] == 2; } // NOLINT
```

Comment:
a) Integer numbers should never be used directly to encode different cases or options. This is what enumerations are for (`enum class`).